src/libraries/System.Private.CoreLib/src/System/Text/CompositeFormat.cs

```diff
@@ -0,0 +1,353 @@
+                    while (char.IsAsciiDigit(ch))
+                    {
+                        index = index * 10 + ch - '0';
+                        if (!TryMoveNext(format, ref pos, out ch))
+                        {
+                            return false;
+                        }
+                    }
+
+                    // Consume optional whitespace.
+                    while (ch == ' ')
+                    {
+                        if (!TryMoveNext(format, ref pos, out ch))
+                        {
+                            return false;
+                        }
+                    }
+
+                    // Parse the optional alignment, which is of the form:
+                    //     comma
+                    //     optional any number of spaces
+                    //     optional -
+                    //     at least one digit
+                    //     optional any number of spaces
+                    if (ch == ',')
+                    {
+                        // Consume optional whitespace.
+                        do
+                        {
+                            if (!TryMoveNext(format, ref pos, out ch))
+                            {
+                                return false;
+                            }
+                        }
+                        while (ch == ' ');
+
+                        // Consume an optional minus sign indicating left alignment.
+                        int leftJustify = 1;
+                        if (ch == '-')
+                        {
+                            leftJustify = -1;
+                            if (!TryMoveNext(format, ref pos, out ch))
+                            {
+                                return false;
+                            }
+                        }
+
+                        // Parse alignment digits. The read character must be a digit.
+                        width = ch - '0';
+                        if ((uint)width >= 10u)
+                        {
+                            return false;
+                        }
+                        if (!TryMoveNext(format, ref pos, out ch))
+                        {
+                            return false;
+                        }
+                        while (char.IsAsciiDigit(ch))
+                        {
+                            width = width * 10 + ch - '0';
+                            if (!TryMoveNext(format, ref pos, out ch))
+                            {
+                                return false;
+                            }
+                        }
+                        width *= leftJustify;
+
+                        // Consume optional whitespace
+                        while (ch == ' ')
+                        {
+                            if (!TryMoveNext(format, ref pos, out ch))
+                            {
+                                return false;
+                            }
+                        }
```

Comment:
Is this worth moving it to inlined method? it is repeated multiple times.

Reply:
It's repeated twice, and then there's another but using this same loop would add an unnecessary comparison.  I'm not sure it's worth it.  This is also a direct copy of the routine that's already used for string.Format, with as few modifications as possible, so I'm hesitant to make any changes to the core logic here.